lib/dialects/abstract/query-generator.js

```diff
@@ -1536,7 +1538,7 @@ var QueryGenerator = {
                    left.rawAttributes[left.primaryKeyAttribute].field
 
       /* Attributes for the right side */
-      , right = association.target
+      , right = include.model
```

Comment:
This honors a models being passed on the include. association.target is the model at the time of the association being created. If you try to do 
```javascript
 return User.findAll({
    where: {
        first_name: 'matt'
    },
    include: [{
        model: Comments.schema('new_schema'),
        as: 'comments'
    }]
}); 
```
without this code it will join the included table with the schema set on base model. Without this .schema is pretty worthless for includes because you would have to add a separate association for every schema.